File: solutions/cpp/luhn/1/luhn.cpp
#include "luhn.h"
using namespace std;
namespace luhn {
    bool valid(string num)  {
        
        string nbnum = "";
        for (size_t i=0; i<num.length();i++){
            if (num.at(i)>='0' && num.at(i)<='9') nbnum+=num.at(i);
            else if (num.at(i)!=' ') return false;          
        }
        if (nbnum.length()<=1) return false;
        int s=0;
        bool even = false;
        for (int i=(int)nbnum.length()-1; i>=0; i--) {
            int d=nbnum.at(i)-'0';
            if (even) {
                d*=2;
                if (d>9) d-=9;
                even = false;
            }
            else even=true;  
            s+=d;  
        }
        if (s % 10 == 0) return true;
        else return false;
    }
}  // namespace luhn

File: solutions/javascript/luhn/1/luhn.js
//
// This is only a SKELETON file for the 'Luhn' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const valid = (num) => {
  num = num.replace(/[ .]/g,'');
  if (num.length <= 1) return false;
  var s = 0;
  var sd = 1;
  var n;
  for (var i=num.length-1; i>=0; i--) {
    if (sd == 2) {
      n = 2 * parseInt(num.substr(i,1));
      if (n > 9) n = n -9;
      sd = 1;
    } else {
      n = parseInt(num.substr(i,1));
      sd = 2;
    }
    s += n;
  }
  if (s % 10 == 0) return true;
  return false;
};
